radix-engine/src/updates/protocol_builder.rs

```diff
@@ -0,0 +1,131 @@
+use radix_substate_store_interface::db_key_mapper::SpreadPrefixKeyMapper;
+
+use super::*;
+
+#[derive(Clone)]
+pub struct ProtocolUpdateExecutor {
+    pub protocol_update: ProtocolUpdate,
+    pub batch_generator: Box<dyn ProtocolUpdateBatchGenerator>,
+}
+
+impl ProtocolUpdateExecutor {
+    pub fn run_and_commit<S: SubstateDatabase + CommittableSubstateDatabase>(self, store: &mut S) {
+        let mut current_batch_index = 0;
+        let batch_generator = self.batch_generator;
+        loop {
+            let Some(batch) = batch_generator.generate_batch(store, current_batch_index) else {
+                break;
+            };
+            for transaction in batch.transactions {
+                let state_updates = match transaction {
+                    ProtocolUpdateTransactionDetails::FlashV1Transaction(flash) => {
+                        flash.state_updates
+                    }
+                };
+                let db_updates = state_updates.create_database_updates::<SpreadPrefixKeyMapper>();
+                store.commit(&db_updates);
+            }
+            current_batch_index += 1;
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct ProtocolBuilder {
+    settings: ProtocolSettings,
+}
+
+#[derive(Clone)]
+pub struct ProtocolSettings {
+    // TODO: It would be nice to move bootstrap / Genesis into this formulation
+    anemone: AnemoneSettings,
+    bottlenose: BottlenoseSettings,
+}
+
+impl ProtocolBuilder {
+    pub fn for_simulator() -> Self {
+        Self::for_network(&NetworkDefinition::simulator())
+    }
+
+    pub fn for_network(network_definition: &NetworkDefinition) -> Self {
+        Self {
+            settings: ProtocolSettings {
+                anemone: AnemoneSettings::all_enabled_as_default_for_network(network_definition),
+                bottlenose: BottlenoseSettings::all_enabled_as_default_for_network(
+                    network_definition,
+                ),
+            },
+        }
+    }
+
+    pub fn with_anemone(mut self, settings: AnemoneSettings) -> Self {
+        self.settings.anemone = settings;
+        self
+    }
+
+    pub fn with_bottlenose(mut self, settings: BottlenoseSettings) -> Self {
+        self.settings.bottlenose = settings;
+        self
+    }
+
+    pub fn until_genesis(self) -> ProtocolExecutor {
```

Comment:
Ditto regarding naming.

Reply:
Thought we reached consensus to rename to `babylon`?

Reply:
Haha I couldn't tell we reached consensus. Will rename it :)